master/internal/api_tasks.go

```diff
@@ -247,14 +247,54 @@ func (a *apiServer) GetTaskAcceleratorData(
 	ctx context.Context,
 	req *apiv1.GetTaskAcceleratorDataRequest,
 ) (*apiv1.GetTaskAcceleratorDataResponse, error) {
-	return nil, grpcutil.UnimplementedError
+	if req.TaskId == "" {
+		return nil, status.Error(codes.InvalidArgument, "task ID missing")
+	}
+	var res []model.AcceleratorData
+	err := db.Bun().NewSelect().
+		ColumnExpr("alloc_acc.*").
+		TableExpr("allocation_accelerators alloc_acc").
+		Join("LEFT JOIN allocations alloc ON alloc_acc.allocation_id = alloc.allocation_id").
+		Where("alloc.task_id = ?", req.TaskId).Scan(ctx, &res)
+	if err != nil {
+		return nil, fmt.Errorf("querying allocation accelerators: %w", err)
+	}
+	var accelerationData []*apiv1.AcceleratorData
+	for _, r := range res {
+		row := r.Proto()
+		accelerationData = append(accelerationData, &row)
+	}
+	return &apiv1.GetTaskAcceleratorDataResponse{AcceleratorData: accelerationData}, nil
 }
 
 func (a *apiServer) PostAllocationAcceleratorData(
 	ctx context.Context,
 	req *apiv1.PostAllocationAcceleratorDataRequest,
 ) (*apiv1.PostAllocationAcceleratorDataResponse, error) {
-	return nil, grpcutil.UnimplementedError
+	if req.AllocationId == "" {
+		return nil, status.Error(codes.InvalidArgument, "allocation ID missing")
+	}
+
+	if err := a.canEditAllocation(ctx, req.AllocationId); err != nil {
+		return nil, err
+	}
+
+	accData := &model.AcceleratorData{
+		ContainerID:      req.AcceleratorData.ContainerId,
+		AllocationID:     model.AllocationID(req.AllocationId),
+		NodeName:         req.AcceleratorData.NodeName,
+		AcceleratorType:  req.AcceleratorData.AcceleratorType,
+		AcceleratorUuids: req.AcceleratorData.AcceleratorUuids,
+	}
+	err := task.DefaultService.SetAcceleratorData(
+		ctx,
+		*accData,
```

Comment:
nit, just don't make it a pointer
```suggestion
		accData,
```

Reply:
Done

Reply:
fyi doesn't look like this got done, but not a big deal